hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/FieldOperations.java

```diff
@@ -409,6 +546,12 @@ public Object readObject(GenericRecord genericRecord, String fieldName) {
         public void writeFieldFromRecordToWriter(DefaultCompactWriter writer, GenericRecord genericRecord, String fieldName) {
             writer.writeDecimal(fieldName, genericRecord.getDecimal(fieldName));
         }
+
+        @Override
+        public void writeJsonFormattedField(StringBuilder stringBuilder, AbstractGenericRecord record, String fieldName) {
+            BigDecimal value = record.getDecimal(fieldName);
+            stringBuilder.append(value == null ? "null" : value.toString());
```

Comment:
```suggestion
            stringBuilder.append(record.getDecimal(fieldName));
```
Minor: This has the same effect